Example/Controllers/ViewController.swift

```diff
@@ -74,39 +74,99 @@ class ViewController: UIViewController {
+        MiniApp.shared(with: Config.getCurrent()).getMiniAppManifest(miniAppId: miniAppInfo.id, miniAppVersion: miniAppInfo.version.versionId) { (result) in
             switch result {
             case .success(let manifestData):
-                self.displayFirstTimeLaunchScreen(metadataResponse: manifestData, miniAppInfo: miniAppInfo)
+                self.displayFirstTimeLaunchScreen(reqPermissions: manifestData.requiredPermissions ?? [], optPermissions: manifestData.optionalPermissions ?? [], miniAppInfo: miniAppInfo)
             case .failure:
                 self.displayAlert(title: NSLocalizedString("error_title", comment: ""), message: NSLocalizedString("error_single_message", comment: ""), dismissController: true)
             }
         }
     }
 
-    func displayFirstTimeLaunchScreen(metadataResponse: MiniAppManifest, miniAppInfo: MiniAppInfo) {
+    func compareMiniAppMetaData(miniAppInfo: MiniAppInfo, completionHandler: @escaping (Result<Bool, Error>) -> Void) {
+        guard let downloadedManifest = MiniApp.shared().getDownloadedManifest(miniAppId: miniAppInfo.id) else {
+            return completionHandler(.success(true))
+        }
+        MiniApp.shared().getMiniAppManifest(miniAppId: miniAppInfo.id, miniAppVersion: miniAppInfo.version.versionId) { (result) in
+            switch result {
+            case .success(let manifestData):
+                if manifestData.requiredPermissions == nil && manifestData.optionalPermissions == nil {
+                    self.displayMiniApp(miniAppInfo: miniAppInfo)
+                } else {
+                    self.checkIfManifestChanged(latestManifest: manifestData, oldManifest: downloadedManifest, miniAppInfo: miniAppInfo)
+                }
+            case .failure(let error):
+                completionHandler(.failure(error))
+            }
+        }
+    }
+
+    func checkIfManifestChanged(latestManifest: MiniAppManifest, oldManifest: MiniAppManifest, miniAppInfo: MiniAppInfo) {
+        let cachedPermissions = MiniApp.shared().getCustomPermissions(forMiniApp: miniAppInfo.id)
+        let cachedAllowedPermissions = cachedPermissions.filter { $0.isPermissionGranted.boolValue == true }
+
+        let requiredPermissions = filterPermissions(permsArray: latestManifest.requiredPermissions ?? [],
+                                                    cachedPermissions: cachedAllowedPermissions)
+        let optionalPermissions = filterPermissions(permsArray: latestManifest.optionalPermissions ?? [],
+                                                    cachedPermissions: cachedPermissions)
+
+        if oldManifest == latestManifest && requiredPermissions.count == 0 {
+            self.displayMiniApp(miniAppInfo: miniAppInfo)
+        } else {
+            if requiredPermissions.count > 0 || optionalPermissions.count > 0 {
+                self.displayFirstTimeLaunchScreen(reqPermissions: requiredPermissions,
```

Comment:
I think the first-time launch screen should display all of the permissions and meta-data again, not just the data that changed.

It might also be nice to label the new permissions with "New" or something, but that's something we can do later I think.

Reply:
Do we really need to display all the permissions? Because the user has already responded to the permissions before right?
For eg., 1.0 version contains **Required** - Username :: **Optional** - Profile, Contact & Device. 

2.0 - **Required** - Username, Profile :: **Optional** - Contact & Device. 

Scenario 1: User agreed to all the permissions
There is a change in meta-data now, should we display all the permissions again?

Scenario 2: User didn't agree to **Profile** perm at 1.0 and launching 2.0 now
There is a change in meta-data now, should we display all the permissions again? or just the profile permission?

Reply:
Yeah, I think we should show the same information on the launch screen in all of these cases. Basically show all information from the manifest: required permissions, optional permissions, and customMetaData.

This is because it could be confusing to the user that it's only showing the new permissions. It makes it seem like the Mini App only needs these new permissions and not the old permissions anymore.

The only difference is that maybe we can show a "New" label next to permissions that have been added (and an "Updated" label next to the `customMetaData` when it's changed)? But I think that's not necessary right now. We should keep it simple for now and can add an improvement for labels later.

Reply:
> Looks good! Just still need to update the user guide with information about `getDownloadedManifest`. I think something similar to what's in the Android PR is okay:
> https://github.com/rakutentech/android-miniapp/pull/271/files#diff-e0502b4de6bd41029198132f0b09748d40b52a794167c81874134ff7a8021447R503

Done